cmd/get-tenant.go

```diff
@@ -0,0 +1,126 @@
+// Package cmd implements fyde-cli commands
+package cmd
+
+/*
+Copyright © 2019 Fyde, Inc. <hello@fyde.com>
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import (
+	"fmt"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/jedib0t/go-pretty/table"
+	"github.com/jedib0t/go-pretty/text"
+	"github.com/spf13/cobra"
+
+	apitenants "github.com/barracuda-cloudgen-access/access-cli/client/tenants"
+	"github.com/barracuda-cloudgen-access/access-cli/models"
+)
+
+// tenantGetCmd represents the get command
+var tenantGetCmd = &cobra.Command{
+	Use:   "get [tenant ID]",
+	Short: "Get tenant",
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+		err := preRunCheckAuth(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		err = preRunFlagChecks(cmd, args)
+		if err != nil {
+			return err
+		}
+
+		if len(args) == 0 && !cmd.Flags().Changed("id") {
+			return fmt.Errorf("missing tenant ID argument")
+		}
+
+		return nil
+	},
+	RunE: func(cmd *cobra.Command, args []string) error {
+		var tenantID string
+		var err error
+		if cmd.Flags().Changed("id") {
+			tenantID, err = cmd.Flags().GetString("id")
+			if err != nil {
+				return err
+			}
+		} else {
+			tenantID = args[0]
+		}
+
+		params := apitenants.NewGetTenantParams()
+		params.SetID(strfmt.UUID(tenantID))
+
+		resp, err := global.Client.Tenants.GetTenant(params, global.AuthWriter)
+		if err != nil {
+			return processErrorResponse(err)
+		}
+
+		tw := tenantBuildTableWriter()
+		tenantTableWriterAppend(tw, *resp.Payload)
+
+		return printListOutputAndError(cmd, resp.Payload, tw, 1, err)
+	},
+}
+
+func tenantBuildTableWriter() table.Writer {
+	tw := table.NewWriter()
+	tw.Style().Format.Header = text.FormatDefault
+	tw.AppendHeader(table.Row{
+		"ID",
+		"Name",
+		"Created At",
+		"Updated At",
+		"Locked At",
+	})
+	tw.SetAlign([]text.Align{
+		text.AlignRight,
+		text.AlignRight,
+		text.AlignLeft,
+		text.AlignLeft,
+		text.AlignLeft,
+	})
+	tw.SetAllowedColumnLengths([]int{36, 30, 30, 30, 30})
+	return tw
+}
+
+func tenantTableWriterAppend(tw table.Writer, tenant models.Tenant) {
+	tw.AppendRow(table.Row{
+		tenant.ID,
+		tenant.Name,
+		tenant.CreatedAt,
+		tenant.UpdatedAt,
+		tenant.LockedAt,
```

Comment:
not sure it is useful to put the locked at in the view. if you're locked this API will fail anyway

Reply:
removed